src/test/integration/integration_tests.py

```diff
@@ -0,0 +1,276 @@
+import unittest
+import requests
+import argparse
+import json
+import sys
+import logging
+from datetime import date, datetime
+from urllib import parse
+from random import randint
+
+
+class integration_tests(unittest.TestCase):
+
+    # Set class variables related to configurations for tests
+    @classmethod
+    def set_configuration_variables(cls, configuration):
+        config_file_raw = open(configuration)
+        config_file_json = json.load(config_file_raw)
+
+        base_url = config_file_json["hostname"] + "v1/hr"
+        cls.departments_url = base_url + "/departments"
+        cls.positions_url = base_url + "/positions"
+        cls.locations_url = base_url + "/locations"
+
+        # This should be coming from a /businesscenters endpoint
+        cls.business_centers = ["AABC", "AMBC", "ASBC", "BEBC",
+                                "CCBO", "FOBC", "HSBC", "UABC"]
+
+        # Only student positions are currently supported.
+        cls.position_type = "student"
+
+        # Minimal number of locations to count for
+        cls.minimal_location_count = 200
+
+        client_id = config_file_json["client_id"]
+        client_secret = config_file_json["client_secret"]
+
+        access_token = cls.get_access_token(
+                config_file_json["token_api"],
+                client_id,
+                client_secret)
+
+        # Set headers and query parameters
+        cls.auth_header = {'Authorization': 'Bearer %s' % access_token}
+
+    # Helper method to get an access token
+    @staticmethod
+    def get_access_token(url, client_id, client_secret):
+        post_data = {
+            'client_id': client_id,
+            'client_secret': client_secret,
+            'grant_type': 'client_credentials'}
+
+        request = requests.post(url, data=post_data)
+        response = request.json()
+
+        return response["access_token"]
+
+    def __make_request(self, url, params=None):
+        return requests.get(url, params=params, headers=self.auth_header)
+
+    def __departments_request(self, business_center):
+        params = {'businessCenter': business_center}
+        return self.__make_request(self.departments_url, params)
+
+    def test_departments(self):
+        for bc in self.business_centers:
+            logging.debug("Testing business center: {}".format(bc))
+            request = self.__departments_request(bc)
+
+            request_elapsed_seconds = request.elapsed.total_seconds()
+            logging.debug("Request took {} second(s)"
+                          .format(request_elapsed_seconds))
+
+            self.assertEqual(request.status_code, 200)
+            self.assertLess(request_elapsed_seconds, 1)
```

Comment:
Since you tested the elapsed seconds for every request, could you put this test case to another function to make this logic reusable?